### PR TITLE
[Merged by Bors] - feat(Util/CountHeartbeats): Add `guard_min_heartbeats` command

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/CountHeartbeats.lean [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Mathlib.Tactic.Ring
+
+set_option linter.style.header false
+
+/-- info: Used 7 heartbeats, which is less than the current maximum of 200000. -/
```
Comment (Vierkantor) [inline test/CountHeartbeats.lean:5]: How flaky is putting an exact number here? See also #18372 :)

File: Mathlib/Util/CountHeartbeats.lean [hunk 2]
```diff
@@ -117,6 +117,31 @@ elab "count_heartbeats " "in" ppLine cmd:command : command => do
...
+           | some j => j.getNat
+           | none => max
+  let start ← IO.getNumHeartbeats
+  try
+    elabCommand (← `(command| set_option maxHeartbeats 0 in $cmd))
+  finally
+    let finish ← IO.getNumHeartbeats
+    let elapsed := (finish - start) / 1000
+    if elapsed < n then
+      logInfo m!"Used {elapsed} heartbeats, which is less than the guarded quorum of {n}."
```
Comment (Vierkantor) [inline Mathlib/Util/CountHeartbeats.lean:143]: I'm not sure "guarded quorum" is going to be clear to anyone seeing this message. Maybe just "minimum"?

**Review by Vierkantor — APPROVED** [review]
LGTM otherwise, thanks!

File: test/CountHeartbeats.lean (hunk 1 above)
Comment (jcommelin) [reply test/CountHeartbeats.lean:5]: I was hoping that I could get a very stable number by picking a very trivial result. In some sense, if this number changes, I wouldn't mind knowing explicitly that it changed.

**Review by Vierkantor — APPROVED** [review]
bors merge

**Pull request closed without merging**